sync-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -522,6 +523,15 @@ protected static SqlParts whereSqlForAndClause(List<Object> clause, String index
                         // This was validated during normalization.
                         List<Object> inList = (List<Object>) negatedPredicate.get(operator);
                         placeholder = placeholdersForInList(inList, sqlParameters);
+                    } else if (operator.equals(MOD)) {
+                        // The predicate map value must be a two element list containing integers
+                        // here.  This was validated during normalization.
+                        List<Integer> modulus = (List<Integer>) negatedPredicate.get(operator);
+                        // Casting the remainder as integer ensures proper processing of the modulo
+                        // operation by SQLite.  If left un-cast, unexpected results occur.
+                        placeholder = String.format("? %s CAST(? AS INTEGER)", operatorMap.get(EQ));
```

Comment:
So, I notice that we don't use this CAST AS INTEGER on the obj-c code: https://github.com/cloudant/CDTDatastore/blob/master/Classes/common/query/CDTQQuerySqlTranslator.m#L429